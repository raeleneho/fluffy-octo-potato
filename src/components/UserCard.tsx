import { Avatar, Box, HStack, Text } from "@chakra-ui/react";
import { capitalizeFirstLetter } from "../helpers/CapitalizeFirstLetter";

interface UserCardProps {
  name: string;
  role: string;
}

function UserCard({ name, role }: UserCardProps) {
  const extractFirstName = (fullName: string) => fullName.split(" ")[0];

  return (
    <HStack spacing={4}>
      <Avatar
        borderRadius="md"
        color="blue.400"
        backgroundColor={"blue.50"}
        name={extractFirstName(name)}
      />
      <Box>
        <Text fontSize="xl">{name}</Text>
        <Text fontSize="md" color="gray.500">
          {capitalizeFirstLetter(role)}
        </Text>
      </Box>
    </HStack>
  );
}

export default UserCard;
