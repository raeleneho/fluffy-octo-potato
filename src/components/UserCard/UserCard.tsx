import { Avatar, Box, HStack, Text } from "@chakra-ui/react";

interface UserCardProps {
  name: string;
  role: string;
}

function UserCard({ name, role }: UserCardProps) {
  const extractFirstInitial = (fullName: string) => fullName.split(" ")[0];

  const capitalizeFirstLetter = (word: string) =>
    word.charAt(0) + word.slice(1).toLowerCase();

  return (
    <HStack spacing={4} data-test-id="user-card">
      <Avatar
        borderRadius="md"
        color="brandBlue.500"
        backgroundColor="brandBlue.100"
        name={extractFirstInitial(name)}
        sx={{
          width: "40px",
          height: "40px",
        }}
      />
      <Box>
        <Text fontSize="lg">{name}</Text>
        <Text fontSize="sm" color="brandGray.400">
          {capitalizeFirstLetter(role)}
        </Text>
      </Box>
    </HStack>
  );
}

export default UserCard;
