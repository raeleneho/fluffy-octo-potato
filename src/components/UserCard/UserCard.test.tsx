import { render } from "@testing-library/react";

import UserCard from "./UserCard";

test("Unchecked Radio", () => {
  const { container } = render(<UserCard name="Rae" role="boss" />);

  expect(container).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
