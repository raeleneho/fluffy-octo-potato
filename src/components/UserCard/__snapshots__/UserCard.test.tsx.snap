// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unchecked Radio 1`] = `
<div>
  <div
    class="chakra-stack css-1mlvmbj"
    data-test-id="user-card"
  >
    <span
      class="chakra-avatar css-f4zlsr"
    >
      <div
        aria-label="Rae"
        class="chakra-avatar__initials css-0"
        role="img"
      >
        R
      </div>
    </span>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1ezsviu"
      >
        Rae
      </p>
      <p
        class="chakra-text css-1f6lyav"
      >
        boss
      </p>
    </div>
  </div>
</div>
`;
